src/main/java/seedu/address/storage/JsonAdaptedCompany.java

```diff
@@ -0,0 +1,103 @@
+package seedu.address.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.company.BillingDate;
+import seedu.address.model.company.Company;
+import seedu.address.model.company.CompanyName;
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Phone;
+
+/**
+ * Jackson-friendly version of <code>Company</code>.
+ */
+public class JsonAdaptedCompany {
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Company's %s field is missing!";
+    private final String name;
+    private final String address;
+    private final String billingDate;
+    private final String phone;
+
+    /**
+     * Constructs a json adapted company with the company details.
+     *
+     * @param name Company name.
+     * @param address Company address.
+     * @param billingDate Company billing date.
+     * @param phone Company phone number.
+     */
+    @JsonCreator
+    public JsonAdaptedCompany(@JsonProperty("name") String name,
+                              @JsonProperty("address") String address,
+                              @JsonProperty("billing date") String billingDate,
+                              @JsonProperty("phone") String phone) {
+        this.name = name;
+        this.address = address;
+        this.billingDate = billingDate;
+        this.phone = phone;
+    }
+
+    /**
+     * Converts a <code>Company</code> into this class for Jackson use.
+     * @param source Company to be converted.
+     */
+    public JsonAdaptedCompany(Company source) {
+        name = source.getName().fullName;
+        address = source.getAddress().value;
+        billingDate = source.getBillingDate().date;
+        phone = source.getPhone().value;
```

Comment:
For consistency sake, I would suggest changing the public variable name of all fields to `value`. So in this case, `fullName` and `date` -> `value`. 

I know `Person`'s `Name` uses `fullName` as its public variable name as well, but to me it does not make sense to name it other than `value` if `Name` class only has one variable anyways. 

(Unless we are thinking of implementing other variables in `Name` class such as `source.getName().nickName`)